src/DotNetty.Buffers/AbstractByteBuffer.cs

```diff
@@ -933,5 +933,100 @@ public string ToString(int index, int length, Encoding encoding)
         {
             return ByteBufferUtil.DecodeString(this, index, length, encoding);
         }
+
+		public int ForEachByte (ByteProcessor processor)
+		{
+			int index = this.ReaderIndex;
+			int length = this.WriterIndex - index;
+			this.EnsureAccessible ();
+
+			return this.ForEachByteAsc0(index, length, processor);
+		}
+
+		public int ForEachByte (int index, int length, ByteProcessor processor)
+		{
+			this.CheckIndex(index, length);
+
+			return this.ForEachByteAsc0(index, length, processor);
+		}
+
+		int ForEachByteAsc0(int index, int length, ByteProcessor processor) {
+			if (processor == null) {
+				throw new NullReferenceException("processor");
+			}
+
+			if (length == 0) {
+				return -1;
+			}
+
+			int endIndex = index + length;
+			int i = index;
+			try {
+				do {
+					if (processor.Process(GetByte(i))) {
+						i ++;
+					} else {
+						return i;
+					}
+				} while (i < endIndex);
+			} catch (Exception e) {
+				// ToDO: Original: PlatformDependent.throwException(e);
+				throw e;
```

Comment:
never do `throw ex;` as you rethrow losing stack trace this way. Either `throw;` or throw a new exception wrapping original one.